### PR TITLE
[mqtt.homeassistant] Add support for Update component

### LINKED ISSUE #14022: [mqtt.homeassistant] HomeAssistant discover error openhab 3.4
after upgrade from 3.3 to 3.4 I get the following error:

`2022-12-19 17:02:51.538 [WARN ] [ssistant.internal.DiscoverComponents] - HomeAssistant discover error: thing update component type is unsupported: update
2022-12-19 17:02:51.566 [WARN ] [ssistant.internal.DiscoverComponents] - HomeAssistant discover error: thing power_on_behavior component type is unsupported: select`

### PR DESCRIPTION
**Pull request by ccutrer**
This component is fairly non-standard - it doesn't add any channels. Instead, it provides several properties to the thing, and also adds a thing configuration allowing you to trigger an OTA update on a Home Assistant device from MainUI.

Fixes #14022, #14881

### CODE REVIEW

**Comment by ccutrer** [comment]
Ping @antroids 🙏

**Comment by ccutrer** [comment]
Rebased (with no changes), then new commit added to fix compilation problems caused by now being on top of #15427

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Update.java [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+        @SerializedName("payload_install")
+        protected @Nullable String payloadInstall;
+    }
+
+    /**
+     * Describes the state payload if it's JSON
+     */
+    public static class ReleaseState {
+        // these are designed to fit in with the default property of firmwareVersion
+        public static String PROPERTY_LATEST_VERSION = "latestFirmwareVersion";
```
Comment (antroids) [inline bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Update.java:79]: Add final

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Update.java [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+        public static String PROPERTY_TITLE = "firmwareTitle";
```
Comment (antroids) [inline bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Update.java:80]: Add final

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Update.java [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+        public static String PROPERTY_RELEASE_SUMMARY = "firmwareSummary";
```
Comment (antroids) [inline bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Update.java:81]: Add final

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Update.java [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+        public static String PROPERTY_RELEASE_URL = "firmwareURL";
```
Comment (antroids) [inline bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Update.java:82]: Add final

**Review by antroids — CHANGES_REQUESTED** [review]
Are these configs changes merged from another PR?

**Comment by ccutrer** [comment]
> Are these .xml configs changes merged from another PR?

No, they're intentionally part of this PR. The "doUpdate" parameter is part of the thing's config description. The XML changes change the config description from being inline in the thing's XML, to be a ref to a separate config description, and then duplicated to have a separate config description that adds the doUpdate parameter. Then all the thing handler has to do to "add" the doUpdate parameter is change the refed config description on the thing from the base one to the one that has the doUpdate parameters, instead of having to generate a new config description on the fly.

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/handler/HomeAssistantThingHandler.java [hunk 5]
```diff
@@ -342,6 +352,26 @@ protected void updateThingStatus(boolean messageReceived, Optional<Boolean> avai
         }
     }
 
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters)
+            throws ConfigValidationException {
+        if (configurationParameters.containsKey("doUpdate")) {
+            configurationParameters = new HashMap<>(configurationParameters);
+            Object value = configurationParameters.remove("doUpdate");
+            if (value instanceof Boolean && ((Boolean) value) == true) {
```
Comment (antroids) [inline bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/handler/HomeAssistantThingHandler.java:361]: Remove comparison with true: `if (value instanceof Boolean doUpdate && doUpdate) {`

**Review by antroids — CHANGES_REQUESTED** [review]

**Review by antroids — APPROVED** [review]
LGTM,
I didn't test it, hope these dynamic thing configs works well on the UI

**Comment by ccutrer** [comment]
Anything pending before this can be merged?

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/resources/OH-INF/config/homeassistant-thing-config.xml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+			<label>MQTT Base Prefix</label>
+			<description>MQTT base prefix</description>
+			<default>homeassistant</default>
+		</parameter>
+	</config-description>
+
+	<config-description uri="thing-type:mqtt:homeassistant-updatable">
+		<parameter-group name="actions">
+			<label>Actions</label>
+			<description></description>
```
Comment (lolodomo) [inline bundles/org.openhab.binding.mqtt.homeassistant/src/main/resources/OH-INF/config/homeassistant-thing-config.xml:23]: Please remove this empty description

**Review by lolodomo — CHANGES_REQUESTED** [review]
Just a small comment

**Comment by lolodomo** [comment]
@ccutrer : please consider my comment before Sunday if you would like that PR being part of OH 4.1.

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Pull request merged**